### PR TITLE
CANInfraredDistanceSensor (ball load sensor) & InnieControl

### PR DESCRIPTION
**Pull request by ajnadel**
This PR:
- starts using `Innie` class
- connects to the ball load sensor
- implements new `RollIn` command for `Innie`
- rumbles xbox controller on ball load

TODO:
- determine ballLoadSensor distance thresholds

### CODE REVIEW

File: src/org/usfirst/frc4904/robot/commands/shooter/RollIn.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.usfirst.frc4904.standard.custom.controllers.Controller;
+
+/**
+ * Continously control the intake roller with a controller,
+ * and rumble the xbox controller when a ball enters the robot
+ * 
+ * @see MotorControl
+ *		
+ */
+public class RollIn extends MotorControl {
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/robot/commands/shooter/RollIn.java]: Please refactor to InnieControl

File: src/org/usfirst/frc4904/robot/subsystems/Shooter.java [hunk 2]
```diff
@@ -13,17 +15,44 @@
...
+	 * @param hood
+	 *        the Hood unit
+	 * @param flywheel
+	 *        the Flywheel unit
+	 * @param distanceSensor
+	 *        the CAN distance sensor
+	 * @param ballLoadSensor
+	 *        the infrared distance CAN sensor that provides ball load values
+	 */
+	public Shooter(RockNRoller rockNRoller, Hood hood, Flywheel flywheel, CANSensor distanceSensor, CANInfraredDistanceSensor ballLoadSensor) {
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/robot/subsystems/Shooter.java]: This should take in a DistanceSensor - implementation agnostic

File: src/org/usfirst/frc4904/robot/subsystems/Shooter.java [hunk 3]
```diff
@@ -13,17 +15,44 @@
...
 	}
 	
+	/**
+	 * Determine if a ball has passed the intake roller by ballLoadSensor intake values.
+	 * 
+	 * @return boolean representing whether or not a ball is inside the robot
+	 * @see ballLoadSensor
+	 */
+	public boolean isBallLoaded() {
+		double value = ballLoadSensor.getDistance();
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/robot/subsystems/Shooter.java]: Either combine this line and the next or refactor `value` to a better name

File: src/org/usfirst/frc4904/robot/RobotMap.java [hunk 4]
```diff
@@ -118,17 +124,18 @@ public RobotMap() {
 		Component.rightWheel.disablePID(); // TODO add encoders
 		Component.chassis = new TankDrive("StrongholdChassis", Component.leftWheel, Component.rightWheel);
 		// Intake
-		Component.intakeRoller = new VelocityEncodedMotor("topIntakeRoller", new AccelerationCap(Component.pdp), new CANEncoder(Port.Sensors.intakeEncoder), new CANTalon(CAN.intakeRoller));
+		Component.intakeRoller = new Innie(new CANTalon(CAN.intakeRoller), new CANEncoder(Port.Sensors.intakeEncoder));
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/robot/RobotMap.java]: Please move the CANTalon argument to the end to be consistent with VelocityEncodedMotor

File: src/org/usfirst/frc4904/robot/subsystems/Shooter.java (hunk 2 above)
Comment (ajnadel) [inline src/org/usfirst/frc4904/robot/subsystems/Shooter.java]: Whoops -- that was stupid/room to grow @carturn.

File: src/org/usfirst/frc4904/robot/RobotMap.java (hunk 4 above)
Comment (ajnadel) [inline src/org/usfirst/frc4904/robot/RobotMap.java]: This was the existing constructor signature, but I'll change it.

File: src/org/usfirst/frc4904/robot/subsystems/Shooter.java [hunk 5]
```diff
@@ -4,28 +4,56 @@
...
+	 * Construct a Shooter subsystem with each individual component.
+	 * 
+	 * @param rockNRoller
+	 *        the RockNRoller unit
+	 * @param hood
+	 *        the Hood unit
+	 * @param flywheel
+	 *        the Flywheel unit
+	 * @param towerDistanceSensor
+	 *        the CAN distance sensor
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/robot/subsystems/Shooter.java:30]: Change this and the next comment - the distance sensors are implementation agnostic

**Comment by c0d3rman** [comment]
This branch will need to be merged into `master`, as `new-motors` has been merged.

**Pull request closed without merging**